src/core/Map.cpp

```diff
@@ -60,6 +61,7 @@ struct Cluster {
     sc2::Point3D center_of_mass;
     sc2::Point3D town_hall_location;
     std::vector<sc2::Point3D> points;
+    std::vector<const sc2::Unit*> resources;
```

Comment:
In terms of used memory it would be simpler to use single vector of std::pair<sc2::Point3D, sc2::Unit>. But it is ok to use the current approach, too, if it makes the code simpler.